src/components/pages/auth/login.tsx

```diff
@@ -107,7 +107,21 @@ export function Login({ oauth }: { oauth?: boolean }) {
 
   async function onLogin(values: SubmitSelfServiceLoginFlowBody) {
     if (!flow?.id) return
-    const originalPreviousPath = sessionStorage.getItem('previousPathname')
+    const redirection = filterUnwantedRedirection(
+      sessionStorage.getItem('previousPathname')
+    )
+
+    function filterUnwantedRedirection(previousPath: string | null) {
```

Comment:
Cool, is this useful for the other auth redirects as well?
(If yes, maybe export it as a helper)